### PR TITLE
Send additional headers to ODS for BLOB interactions

### PR DESCRIPTION
**Pull request by lagalbra**
Remove SAS token from Blob URL before notifying ODS

@Microsoft/omsagent-devs @haitch

### CODE REVIEW

**Comment by lagalbra** [comment]
Unit and system tests pass. pBuild running now.

File: source/code/plugins/out_oms_blob.rb [hunk 1]
```diff
@@ -222,24 +233,50 @@ def commit_blocks(uri, blocks_committed, blocks_uncommitted, file_path)
...
         "DataType" => "BLOB_UPLOAD_NOTIFICATION",
         "IPName" => "",
         "DataItems" => [
           {
-            "BlobUrl" => uri.to_s,
-            "OriginalDataTypeId" => data_type_id
+            "BlobUrl" => blob_url,
+            "OriginalDataTypeId" => data_type_id,
+            "StartOffset" => offset_blob_size,
+            "FileSize" => sent_size,
```
Comment (haitch) [inline source/code/plugins/out_oms_blob.rb]: this FileSize should be full blobSize, in this case, it should be offset_blob_size + sent_size.

File: source/code/plugins/out_oms_blob.rb [hunk 2]
```diff
@@ -222,24 +233,50 @@ def commit_blocks(uri, blocks_committed, blocks_uncommitted, file_path)
...
+    #   sent_size: int. Amount of data we appended to the BLOB
+    #   etag: string. ETag from the BLOB for this data
+    def notify_blob_upload_complete(uri, data_type, custom_data_type, offset_blob_size, sent_size, etag)
       data_type_id = data_type
       if !custom_data_type.nil?
         data_type_id = "#{data_type}.#{custom_data_type}"
       end
 
+      # Remove SAS token from the URL
+      blob_url = uri.to_s.split("?")[0]
```
Comment (haitch) [inline source/code/plugins/out_oms_blob.rb]: If uri is an strong type object, not sure if you have something similar to this Uri.GetLeftPartial [https://msdn.microsoft.com/en-us/library/system.uri.getleftpart(v=vs.110).aspx].

File: source/code/plugins/out_oms_blob.rb [hunk 3]
```diff
@@ -54,9 +54,10 @@ def shutdown
     #   uri: URI. blob URI
     #   msg: string. body of the request
     #   file_path: string. file path
+    #   get_etag: bool. True if we should request the ETag
     # returns:
     #   HTTPRequest. blob PUT request
-    def create_blob_put_request(uri, msg, request_id, file_path = nil)
+    def create_blob_put_request(uri, msg, request_id, file_path = nil, get_etag = false)
```
Comment (lagalbra) [inline source/code/plugins/out_oms_blob.rb]: Will change.

File: source/code/plugins/out_oms_blob.rb (hunk 2 above)
Comment (lagalbra) [inline source/code/plugins/out_oms_blob.rb]: Have found a solution here: https://stackoverflow.com/questions/10410523/removing-a-part-of-a-url-with-ruby

Will change to use `uri.fragment = uri.query = nil`

File: source/code/plugins/out_oms_blob.rb (hunk 1 above)
Comment (lagalbra) [inline source/code/plugins/out_oms_blob.rb]: Will change.

**Comment by lagalbra** [comment]
@haitch Please review again.

**Comment by lagalbra** [comment]
@Microsoft/omsagent-devs Please review. This is has been verified locally.

**Pull request merged**